src/main/kotlin/io/github/mojira/arisa/infrastructure/jira/Mappers.kt

```diff
@@ -0,0 +1,155 @@
+package io.github.mojira.arisa.infrastructure.jira
+
+import arrow.core.Either
+import arrow.core.left
+import arrow.core.right
+import arrow.syntax.function.partially1
+import arrow.syntax.function.pipe
+import com.uchuhimo.konf.Config
+import io.github.mojira.arisa.domain.Attachment
+import io.github.mojira.arisa.domain.ChangeLogItem
+import io.github.mojira.arisa.domain.Comment
+import io.github.mojira.arisa.domain.Link
+import io.github.mojira.arisa.domain.LinkParam
+import io.github.mojira.arisa.domain.LinkedIssue
+import io.github.mojira.arisa.domain.Version
+import io.github.mojira.arisa.infrastructure.config.Arisa
+import net.rcarz.jiraclient.JiraClient
+import net.sf.json.JSONObject
+import java.text.SimpleDateFormat
+import net.rcarz.jiraclient.Attachment as JiraAttachment
+import net.rcarz.jiraclient.ChangeLogEntry as JiraChangeLogEntry
+import net.rcarz.jiraclient.ChangeLogItem as JiraChangeLogItem
+import net.rcarz.jiraclient.Comment as JiraComment
+import net.rcarz.jiraclient.Issue as JiraIssue
+import net.rcarz.jiraclient.IssueLink as JiraIssueLink
+import net.rcarz.jiraclient.Project as JiraProject
+import net.rcarz.jiraclient.Version as JiraVersion
+
+private val isoFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ")
+private fun String.toInstant() = isoFormat.parse(this).toInstant()
+
+private fun JiraIssue.getFieldAsString(field: String) = this.getField(field) as? String?
+
+private fun JiraIssue.getCustomField(customField: String): String? =
+    ((getField(customField)) as? JSONObject)?.get("value") as? String?
+
+fun JiraIssue.getEnvironment() = getFieldAsString("environment")
+
+fun JiraIssue.getCHK(config: Config) = getFieldAsString(config[Arisa.CustomFields.chkField])
+fun JiraIssue.getConfirmation(config: Config) = getCustomField(config[Arisa.CustomFields.confirmationField])
```

Comment:
There are several duplicated codes of converting `null` or empty confirmation status to `Unconfirmed` (like the `String#getOrDefault` extension function in `RevokeConfirmationModule`).

Will it make sense to move that conversion here?

Reply:
(Example: A module that sets to Unconfirmed when it's set to null or empty)